### DISCUSSION TITLE
Target class dependencies causing  java.lang.NoClassDefFoundError

### DISCUSSION THREAD

**Discussion by dmgping** [CodeIntelligenceTesting/jazzer#692] [General]
Attempting to run the following, 

`docker run -v /project-path:/fuzzing cifuzz/jazzer --cp=project.jar --target_class=org.example.project.Fuzzer`

when my class being fuzzed inside the Fuzzer class relies on log4j results in the following exception:
`Java Exception: java.lang.NoClassDefFoundError: org/apache/logging/log4j/core/StringLayout`

Is there a way to include all dependencies in the mounted volume/jar to overcome this issue? or should I be using
`--autofuzz_ignore=java.lang.NoClassDefFoundError` I'm unsure if my fuzz target is actually getting tested because of the error.

**Comment by centic9** [comment]
You have to include all the class-files from all third-party libraries in one jar, otherwise your fuzzing will not work via docker. Ignoring the exception would likely lead to no fuzzing of your code at all.

A good way is to use the "shadow" plugin in Gradle.

See https://github.com/centic9/demo-fuzz/ for a sample project which contains everything in the build.gradle and usage-instruction in the README.
